### ISSUE TITLE
Collect payment method for setup intent doesn't update its `status`

### ISSUE THREAD

**Issue by TatsuUkraine** [stripe/stripe-terminal-android#449]
## Summary
collectSetupIntentPaymentMethod completes with the setup intent that has requires payment method status instead of requires confirmation. Not quite sure if it's expected behavior for sandbox mode.

## Android version
Android 14

## Impacted devices (Android devices or readers)
All

## SDK version
3.4.0

**Comment by billfinn-stripe** [comment]
Hi @TatsuUkraine -- you're correct that the `status` is incorrect, but the payment method is, in fact, collected.

We do have plans to fix this, but I don't have a timeline right now.

Can you share how you're using the `status` field currently in your integration?

**Comment by TatsuUkraine** [comment]
@billfinn-stripe hi, thank you for quick answer. I wanted to optimize payment collect process from direct method calls to reactive state listeners. To automatically perform setup intent/payment intent confirmation whenever status changes in the state. Since I thought it would be more reliable than making an assumption that if payment method collect call completed successfully then it's ready for confirmation

**Comment by billfinn-stripe** [comment]
Got it, thanks. Yes, the status is correct for Payment Intents, but we don't update it for Setup Intents.

For the immediate term, I think you'll have to rely on the successful result of Setup Intent collection to move on to Setup Intent confirmation.

I'll leave this Github issue open until we've addressed the underlying status inconsistency.

**Comment by TatsuUkraine** [comment]
Thank you!

**Comment by TatsuUkraine** [comment]
@billfinn-stripe hello, I have a follow-up question. Is that ok that after a successful payment method collect, setup intent doesn't have the payment method attached to it?

**Comment by TatsuUkraine** [comment]
Ok, is that possible to confirm setup intent on the backend (using php sdk) if payment method was collected through the terminal sdk?

**Comment by billfinn-stripe** [comment]
No, `confirmSetupIntent` only exists within the Terminal SDKs, and while the payment method ID is not set on the Setup Intent, the object does, in fact, have the payment method details that were collected from the card.

**Comment by TatsuUkraine** [comment]
there is confirm action in PHP sdk, that requires payment method to proceed, so I'm curious if it's possible in general to collect payment method with SDK, and then confirm the setup intent using [PHP sdk](https://docs.stripe.com/api/setup_intents/confirm)

**Comment by billfinn-stripe** [comment]
Ah, I see -- thanks. The `/confirm` API endpoint you are referencing is meant to be used for card-not-present (i.e., ecommerce) transactions. For card-present transactions, it's not possible to collect payment method details via the Terminal SDK and then confirm the setup intent using the `/confirm` API.

**Comment by TatsuUkraine** [comment]
Ah, ok, thank you)
